library/objective-c/EnvoyEngineImpl.m

```diff
@@ -95,13 +44,68 @@ static envoy_headers toNativeHeaders(EnvoyHeaders *headers) {
   }
 
   EnvoyHeaders *platformHeaders = to_ios_headers(headers);
-  release_envoy_headers(headers);
```

Comment:
we were always releasing right after the transformation call, right? Now I can't remember why https://github.com/lyft/envoy-mobile/pull/971/files#diff-0fdd94f3084afc47d1b505544703ca0dL41 existed in the first place? i.e there was not a site (is there one in this PR and I am blind to it) where we conditionally released?

Reply:
It's already gone. I removed the optimization and came up with the flag-based way (that we discussed earlier) to do it (later).